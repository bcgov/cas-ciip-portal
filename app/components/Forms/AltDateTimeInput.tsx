// https://github.com/rjsf-team/react-jsonschema-form/blob/v1.8.1/src/components/widgets/AltDateTimeWidget.js

import React from "react";
import PropTypes from "prop-types";
import AltDateInput from "./AltDateInput";
import { WidgetProps } from "react-jsonschema-form";

function AltDateTimeInput(props: WidgetProps) {
  const { AltDateInput } = props.registry.widgets;
  return <AltDateInput time {...props} />;
}

if (process.env.NODE_ENV !== "production") {
  AltDateTimeInput.propTypes = {
    schema: PropTypes.object.isRequired,
    id: PropTypes.string.isRequired,
    value: PropTypes.string,
    required: PropTypes.bool,
    onChange: PropTypes.func,
    options: PropTypes.object,
  };
}

AltDateTimeInput.defaultProps = {
  ...AltDateInput.defaultProps,
  time: true,
};

export default AltDateTimeInput;