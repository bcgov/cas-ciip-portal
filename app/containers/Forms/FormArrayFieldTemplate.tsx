import React from 'react';
import {ArrayFieldTemplateProps} from 'react-jsonschema-form';
import {Form, Button, Col} from 'react-bootstrap';

const FormArrayFieldTemplate: React.FunctionComponent<ArrayFieldTemplateProps> = ({
  items,
  canAdd,
  onAddClick,
  uiSchema,
  className
}) => {
  return (
    <div className={className}>
      {items.map(element => {
        return (
          <React.Fragment key={element.index}>
            <Form.Row>
              <Col xs={12} md={element.hasRemove ? 10 : 12}>
                {element.children}
              </Col>

              {element.hasRemove && (
                <Col xs={12} md={2}>
                  <div className="remove-button-container">
                    <Button
                      type="button"
                      variant="danger"
                      onClick={element.onDropIndexClick(element.index)}
                    >
                      {uiSchema['ui:remove-text'] || 'Remove'}
                    </Button>
                  </div>
                </Col>
              )}
            </Form.Row>
            <hr />
          </React.Fragment>
        );
      })}
      {canAdd && (
        <Form.Row>
          <Col xs={12}>
            <div className="add-button-container">
              <Button type="button" onClick={onAddClick}>
                {uiSchema['ui:add-text'] || 'Add'}
              </Button>
            </div>
          </Col>
        </Form.Row>
      )}
      <style jsx>
        {`
          .remove-button-container {
            height: 100%;
            display: flex;
            justify-content: space-around;
            align-items: center;
          }
          .add-button-container {
            height: 100%;
            display: flex;
            justify-content: space-around;
            align-items: center;
          }
        `}
      </style>
    </div>
  );
};

export default FormArrayFieldTemplate;
