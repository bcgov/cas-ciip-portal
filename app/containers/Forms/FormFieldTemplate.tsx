import React from 'react';
import {Col, Form} from 'react-bootstrap';
import {FieldTemplateProps} from 'react-jsonschema-form';

const FormFieldTemplate: React.FunctionComponent<FieldTemplateProps> = ({
  id,
  label,
  help,
  required,
  description,
  errors,
  children,
  schema,
  classNames,
  uiSchema,
  hidden
}) => {
  if (hidden) return <div className="hidden">{children}</div>;

  const helpText = uiSchema['ui:helplink'] ? (
    <a
      className="font-italic"
      href={uiSchema['ui:helplink']}
      target="_blank"
      rel="noopener noreferrer"
    >
      {help}
    </a>
  ) : (
    <span className="font-italic text-muted">{help}</span>
  );

  if (schema.type === 'array')
    return (
      <Col xs={12}>
        {description}
        {helpText}
        {errors}
        {children}
      </Col>
    );
  if (schema.type === 'object')
    return (
      <Col xs={12}>
        <Form.Row>
          {description}
          {errors}
          {helpText}
          {children}
        </Form.Row>
      </Col>
    );

  const xs = uiSchema['ui:col-xs'] || 12;
  const sm = uiSchema['ui:col-sm'] || xs;
  const md = uiSchema['ui:col-md'] || (sm < 6 ? sm : 6);
  const lg = uiSchema['ui:col-lg'] || md;
  const xl = uiSchema['ui:col-xl'] || lg;

  return (
    <Form.Group
      as={Col}
      xs={xs}
      sm={sm}
      md={md}
      lg={lg}
      xl={xl}
      className={classNames}
    >
      <Form.Label htmlFor={id}>
        {label}
        {required ? <>&nbsp;*</> : null}
      </Form.Label>
      {description}
      {children}
      {errors}
      {helpText}
    </Form.Group>
  );
};

export default FormFieldTemplate;
