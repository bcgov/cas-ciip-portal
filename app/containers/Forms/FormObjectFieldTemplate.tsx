import React from 'react';
import {Col} from 'react-bootstrap';

const FormObjectFieldTemplate = ({
  TitleField,
  properties,
  title,
  description
}) => {
  return (
    <>
      {title && (
        <Col xs={12}>
          <TitleField title={title} />
        </Col>
      )}
      {properties.map(prop => (
        <React.Fragment key={prop.content.key}>{prop.content}</React.Fragment>
      ))}
      {description}
    </>
  );
};

export default FormObjectFieldTemplate;
