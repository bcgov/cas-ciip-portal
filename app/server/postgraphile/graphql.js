const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
