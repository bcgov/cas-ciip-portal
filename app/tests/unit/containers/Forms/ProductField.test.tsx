import React from 'react';
import {shallow} from 'enzyme';
import {ProductFieldComponent} from 'containers/Forms/ProductField';
import productionSchema from 'schema/data/prod/form_json/production.json';
import {JSONSchema7} from 'json-schema';
import {getDefaultRegistry} from '@rjsf/core/dist/cjs/utils';
import FormArrayFieldTemplate from 'containers/Forms/FormArrayFieldTemplate';
import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
import {ProductField_query} from '__generated__/ProductField_query.graphql';
import {ProductField_naicsCode} from '__generated__/ProductField_naicsCode.graphql';

describe('The ProductionFields Component with published product matched to a naics code', () => {
  const query: ProductField_query = {
    ' $refType': 'ProductField_query',
    allProducts: {
      edges: [
        {
          node: {
            rowId: 1,
            units: 'bar',
            productState: 'PUBLISHED',
            requiresEmissionAllocation: true,
            requiresProductAmount: true,
            isEnergyProduct: false,
            addPurchasedElectricityEmissions: false,
            addPurchasedHeatEmissions: false,
            subtractExportedElectricityEmissions: false,
            subtractExportedHeatEmissions: false,
            subtractGeneratedElectricityEmissions: false,
            subtractGeneratedHeatEmissions: false,
            addEmissionsFromEios: false
          }
        }
      ]
    }
  };
  const naicsCode: ProductField_naicsCode = {
    ' $refType': 'ProductField_naicsCode',
    id: 'abc',
    allProductsByNaicsCode: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'bar',
            productState: 'PUBLISHED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: {
      productRowId: 1,
      productUnits: 'bar',
      productEmissions: 123,
      productAmount: 456,
      requiresEmissionAllocation: true
    },
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot when rendering a product', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render any warnings', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper.exists('Alert')).toBe(false);
  });
});

describe('The ProductionFields Component with published product not matched to a naics code', () => {
  const query: ProductField_query = {
    ' $refType': 'ProductField_query',
    allProducts: {
      edges: [
        {
          node: {
            rowId: 1,
            units: 'bar',
            productState: 'PUBLISHED',
            requiresEmissionAllocation: true,
            requiresProductAmount: true,
            isEnergyProduct: false,
            addPurchasedElectricityEmissions: false,
            addPurchasedHeatEmissions: false,
            subtractExportedElectricityEmissions: false,
            subtractExportedHeatEmissions: false,
            subtractGeneratedElectricityEmissions: false,
            subtractGeneratedHeatEmissions: false,
            addEmissionsFromEios: false
          }
        }
      ]
    }
  };
  const naicsCode: ProductField_naicsCode = {
    ' $refType': 'ProductField_naicsCode',
    id: 'abc',
    allProductsByNaicsCode: {
      edges: [
        {
          node: {
            rowId: 2,
            productName: 'baz',
            productState: 'PUBLISHED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: {
      productRowId: 1,
      productUnits: 'bar',
      productEmissions: 123,
      productAmount: 456,
      requiresEmissionAllocation: true
    },
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot when rendering a product not matched to a naics code', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the not in naics code warning', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper.find('Alert').text()).toContain(
      'Product or Service is not associated with the NAICS code'
    );
  });
});

describe('The ProductionFields Component with archived product matched to a naics code', () => {
  const query: ProductField_query = {
    ' $refType': 'ProductField_query',
    allProducts: {
      edges: [
        {
          node: {
            rowId: 1,
            units: 'bar',
            productState: 'ARCHIVED',
            requiresEmissionAllocation: true,
            requiresProductAmount: true,
            isEnergyProduct: false,
            addPurchasedElectricityEmissions: false,
            addPurchasedHeatEmissions: false,
            subtractExportedElectricityEmissions: false,
            subtractExportedHeatEmissions: false,
            subtractGeneratedElectricityEmissions: false,
            subtractGeneratedHeatEmissions: false,
            addEmissionsFromEios: false
          }
        }
      ]
    }
  };

  const naicsCode: ProductField_naicsCode = {
    ' $refType': 'ProductField_naicsCode',
    id: 'abc',
    allProductsByNaicsCode: {
      edges: [
        {
          node: {
            rowId: 1,
            productName: 'bar',
            productState: 'ARCHIVED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: {
      productRowId: 1,
      productUnits: 'bar',
      productEmissions: 123,
      productAmount: 456,
      requiresEmissionAllocation: true
    },
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot when rendering an archived product', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render the archived warning', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper.find('Alert').text()).toContain(
      'Product or Service has been archived'
    );
  });
});

describe('The ProductionFields Component with archived product not matched to a naics code', () => {
  const query: ProductField_query = {
    ' $refType': 'ProductField_query',
    allProducts: {
      edges: [
        {
          node: {
            rowId: 1,
            units: 'bar',
            productState: 'ARCHIVED',
            requiresEmissionAllocation: true,
            requiresProductAmount: true,
            isEnergyProduct: false,
            addPurchasedElectricityEmissions: false,
            addPurchasedHeatEmissions: false,
            subtractExportedElectricityEmissions: false,
            subtractExportedHeatEmissions: false,
            subtractGeneratedElectricityEmissions: false,
            subtractGeneratedHeatEmissions: false,
            addEmissionsFromEios: false
          }
        }
      ]
    }
  };

  const naicsCode: ProductField_naicsCode = {
    ' $refType': 'ProductField_naicsCode',
    id: 'abc',
    allProductsByNaicsCode: {
      edges: [
        {
          node: {
            rowId: 2,
            productName: 'baz',
            productState: 'ARCHIVED'
          }
        }
      ]
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode,
    formData: {
      productRowId: 1,
      productUnits: 'bar',
      productEmissions: 123,
      productAmount: 456,
      requiresEmissionAllocation: true
    },
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot when rendering an archived product not matched to a naics code', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the archived warning and the not in naics warning', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper.find('Alert').at(0).text()).toContain(
      'Product or Service has been archived'
    );
    expect(wrapper.find('Alert').at(1).text()).toContain(
      'Product or Service is not associated with the NAICS code'
    );
  });
});

describe('The ProductionFields Component with no naics code', () => {
  const query: ProductField_query = {
    ' $refType': 'ProductField_query',
    allProducts: {
      edges: []
    }
  };

  const idSchema: any = {$id: 'product_0'};
  const props = {
    schema: productionSchema.schema.definitions.product as JSONSchema7,
    uiSchema: productionSchema.uiSchema,
    idSchema,
    autofocus: false,
    disabled: false,
    errorSchema: null,
    formContext: {},
    readonly: false,
    required: false,
    registry: {
      ...getDefaultRegistry(),
      ArrayFieldTemplate: FormArrayFieldTemplate,
      ObjectFieldTemplate: FormObjectFieldTemplate,
      FieldTemplate: FormFieldTemplate
    } as any,

    name: 'product',
    query,
    naicsCode: null,
    formData: {},
    onChange: jest.fn(),
    onBlur: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the no products found warning', () => {
    const wrapper = shallow(<ProductFieldComponent {...props} />);
    expect(wrapper.find('Alert').at(0).text()).toContain(
      'No products were found'
    );
  });
});
