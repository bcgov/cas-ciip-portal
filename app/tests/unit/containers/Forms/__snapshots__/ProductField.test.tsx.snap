// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ProductionFields Component with archived product matched to a naics code should match the snapshot when rendering an archived product 1`] = `
<Fragment>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     This version of the Product or Service has been archived. This archived product should be removed and an appropriate replacement selected (it may have the same name).
  </Alert>
  <ObjectField
    autofocus={false}
    disabled={true}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "productAmount": 456,
        "productEmissions": 123,
        "productRowId": 1,
        "productUnits": "bar",
        "requiresEmissionAllocation": true,
      }
    }
    idSchema={
      Object {
        "$id": "myField",
      }
    }
    naicsCode={
      Object {
        " $refType": "ProductField_naicsCode",
        "allProductsByNaicsCode": Object {
          "edges": Array [
            Object {
              "node": Object {
                "productState": "ARCHIVED",
                "rowId": 1,
              },
            },
          ],
        },
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "ProductField_query",
        "allProducts": Object {
          "edges": Array [
            Object {
              "node": Object {
                "addEmissionsFromEios": false,
                "addPurchasedElectricityEmissions": false,
                "addPurchasedHeatEmissions": false,
                "isEnergyProduct": false,
                "productState": "ARCHIVED",
                "requiresEmissionAllocation": true,
                "requiresProductAmount": true,
                "rowId": 1,
                "subtractExportedElectricityEmissions": false,
                "subtractExportedHeatEmissions": false,
                "subtractGeneratedElectricityEmissions": false,
                "subtractGeneratedHeatEmissions": false,
                "units": "bar",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The ProductionFields Component with archived product not matched to a naics code should match the snapshot when rendering an archived product not matched to a naics code 1`] = `
<Fragment>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     This version of the Product or Service has been archived. This archived product should be removed and an appropriate replacement selected (it may have the same name).
  </Alert>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     This Product or Service is not associated with the NAICS code reported in this application. Please review the guidance documents for a list of the valid products for this sector or verify the NAICS code reported in the administrative data is correct.
  </Alert>
  <ObjectField
    autofocus={false}
    disabled={true}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "productAmount": 456,
        "productEmissions": 123,
        "productRowId": 1,
        "productUnits": "bar",
        "requiresEmissionAllocation": true,
      }
    }
    idSchema={
      Object {
        "$id": "myField",
      }
    }
    naicsCode={
      Object {
        " $refType": "ProductField_naicsCode",
        "allProductsByNaicsCode": Object {
          "edges": Array [
            Object {
              "node": Object {
                "productState": "ARCHIVED",
                "rowId": 2,
              },
            },
          ],
        },
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "ProductField_query",
        "allProducts": Object {
          "edges": Array [
            Object {
              "node": Object {
                "addEmissionsFromEios": false,
                "addPurchasedElectricityEmissions": false,
                "addPurchasedHeatEmissions": false,
                "isEnergyProduct": false,
                "productState": "ARCHIVED",
                "requiresEmissionAllocation": true,
                "requiresProductAmount": true,
                "rowId": 1,
                "subtractExportedElectricityEmissions": false,
                "subtractExportedHeatEmissions": false,
                "subtractGeneratedElectricityEmissions": false,
                "subtractGeneratedHeatEmissions": false,
                "units": "bar",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The ProductionFields Component with no naics code should match the snapshot 1`] = `
<Fragment>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     No products were found matching the reported NAICS code. Please verify the NAICS code reported in the administrative data.
  </Alert>
  <ObjectField
    autofocus={false}
    disabled={false}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={Object {}}
    idSchema={
      Object {
        "$id": "myField",
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "ProductField_query",
        "allProducts": Object {
          "edges": Array [],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The ProductionFields Component with published product matched to a naics code should match the snapshot when rendering a product 1`] = `
<Fragment>
  <ObjectField
    autofocus={false}
    disabled={false}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "productAmount": 456,
        "productEmissions": 123,
        "productRowId": 1,
        "productUnits": "bar",
        "requiresEmissionAllocation": true,
      }
    }
    idSchema={
      Object {
        "$id": "myField",
      }
    }
    naicsCode={
      Object {
        " $refType": "ProductField_naicsCode",
        "allProductsByNaicsCode": Object {
          "edges": Array [
            Object {
              "node": Object {
                "productState": "PUBLISHED",
                "rowId": 1,
              },
            },
          ],
        },
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "ProductField_query",
        "allProducts": Object {
          "edges": Array [
            Object {
              "node": Object {
                "addEmissionsFromEios": false,
                "addPurchasedElectricityEmissions": false,
                "addPurchasedHeatEmissions": false,
                "isEnergyProduct": false,
                "productState": "PUBLISHED",
                "requiresEmissionAllocation": true,
                "requiresProductAmount": true,
                "rowId": 1,
                "subtractExportedElectricityEmissions": false,
                "subtractExportedHeatEmissions": false,
                "subtractGeneratedElectricityEmissions": false,
                "subtractGeneratedHeatEmissions": false,
                "units": "bar",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;

exports[`The ProductionFields Component with published product not matched to a naics code should match the snapshot when rendering a product not matched to a naics code 1`] = `
<Fragment>
  <Alert
    closeLabel="Close alert"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="danger"
  >
    <strong>
      Warning:
    </strong>
     This Product or Service is not associated with the NAICS code reported in this application. Please review the guidance documents for a list of the valid products for this sector or verify the NAICS code reported in the administrative data is correct.
  </Alert>
  <ObjectField
    autofocus={false}
    disabled={true}
    errorSchema={Object {}}
    formContext={Object {}}
    formData={
      Object {
        "productAmount": 456,
        "productEmissions": 123,
        "productRowId": 1,
        "productUnits": "bar",
        "requiresEmissionAllocation": true,
      }
    }
    idSchema={
      Object {
        "$id": "myField",
      }
    }
    naicsCode={
      Object {
        " $refType": "ProductField_naicsCode",
        "allProductsByNaicsCode": Object {
          "edges": Array [
            Object {
              "node": Object {
                "productState": "PUBLISHED",
                "rowId": 2,
              },
            },
          ],
        },
      }
    }
    name=""
    onBlur={[MockFunction]}
    onChange={[Function]}
    query={
      Object {
        " $refType": "ProductField_query",
        "allProducts": Object {
          "edges": Array [
            Object {
              "node": Object {
                "addEmissionsFromEios": false,
                "addPurchasedElectricityEmissions": false,
                "addPurchasedHeatEmissions": false,
                "isEnergyProduct": false,
                "productState": "PUBLISHED",
                "requiresEmissionAllocation": true,
                "requiresProductAmount": true,
                "rowId": 1,
                "subtractExportedElectricityEmissions": false,
                "subtractExportedHeatEmissions": false,
                "subtractGeneratedElectricityEmissions": false,
                "subtractGeneratedHeatEmissions": false,
                "units": "bar",
              },
            },
          ],
        },
      }
    }
    readonly={false}
    registry={
      Object {
        "definitions": Object {},
        "fields": Object {
          "AnyOfField": [Function],
          "ArrayField": [Function],
          "BooleanField": [Function],
          "DescriptionField": [Function],
          "NullField": [Function],
          "NumberField": [Function],
          "ObjectField": undefined,
          "OneOfField": [Function],
          "SchemaField": [Function],
          "StringField": [Function],
          "TitleField": [Function],
          "UnsupportedField": [Function],
        },
        "formContext": Object {},
        "rootSchema": Object {},
        "widgets": Object {
          "AltDateTimeWidget": [Function],
          "AltDateWidget": [Function],
          "BaseInput": [Function],
          "CheckboxWidget": [Function],
          "CheckboxesWidget": [Function],
          "ColorWidget": [Function],
          "DateTimeWidget": [Function],
          "DateWidget": [Function],
          "EmailWidget": [Function],
          "FileWidget": [Function],
          "HiddenWidget": [Function],
          "PasswordWidget": [Function],
          "RadioWidget": [Function],
          "RangeWidget": [Function],
          "SelectWidget": [Function],
          "TextWidget": [Function],
          "TextareaWidget": [Function],
          "URLWidget": [Function],
          "UpDownWidget": [Function],
        },
      }
    }
    required={false}
    schema={Object {}}
    uiSchema={Object {}}
  />
</Fragment>
`;
